### ISSUE TITLE
FR: official binary release

### ISSUE THREAD

**Issue by andrewayoub** [valeriansaliou/sonic#167]
installing sonic using cargo takes much time, It'd be easier if you have an official binary release

**Comment by valeriansaliou** [comment]
For now, it's impossible to provide a cross-platform binary, as it's not possible to build a MUSL binary due to RocksDB and other deps.

**Comment by andrewayoub** [comment]
Thanks for your quick reply, it doesn't have to be `cross-platform` even providing binaries for specific distributions of linux will be very helpful.

**Comment by valeriansaliou** [comment]
If you have any automated build systems for major Linux distributions and versions to recommend, and even a PR, I'll go for it (yeah, it even depends on the version, eg a Debian 8 build won't work on Debian 9 due to the dyn library links).

**Comment by Mstrodl** [comment]
Have you tried [gitlab-ci](https://gitlab.com/Mstrodl/sonic-builds)? My config seems to work well enough:
* [stretch](https://gitlab.com/Mstrodl/sonic-builds/-/jobs/artifacts/master/raw/target/release/sonic?job=build:stretch)
* [buster](https://gitlab.com/Mstrodl/sonic-builds/-/jobs/artifacts/master/raw/target/release/sonic?job=build:buster)
* [bionic](https://gitlab.com/Mstrodl/sonic-builds/-/jobs/artifacts/master/raw/target/release/sonic?job=build:bionic)

**Comment by Mstrodl** [comment]
( cc @valeriansaliou )

**Comment by valeriansaliou** [comment]
Looks like you're releasing for multiple target platforms. For the sake of simplicity when I do releases, I'd like to be able to build a single MUSL binary that targets all Linux platforms at once. This could be easily possible w/ the latest RocksDB release, which supports static linking.

Tracked in #128

**Comment by Mstrodl** [comment]
@valeriansaliou yeah, but you had said here that (for now) we need per-distro builds

**Comment by valeriansaliou** [comment]
That will not be needed anymore thanks to #128 ; coming soon.

**Comment by RemiBardon** [comment]
I recently make improvements to the CI (#348, #351) and we now have better tooling for releasing to multiple architectures. I only added support for RHEL9 and RHEL10 following a PR (#349), but we can easily add more.

In a future release I will make sure we support `cargo binstall` (see [cargo-binstall/SUPPORT.md in cargo-bins/cargo-binstall](https://github.com/cargo-bins/cargo-binstall/blob/main/SUPPORT.md)) so you can install Sonic even more easily.

Edit: I just checked, `cargo binstall sonic-server` (`cargo binstall sonic-server@1.5.1`) installs from source, meaning `cargo binstall` isn’t able to pick up the Prose archives with its default configuration.
